serde_derive/src/de.rs

```diff
@@ -1965,7 +1965,7 @@ fn deserialize_generated_identifier(
     quote_block! {
         #[allow(non_camel_case_types)]
         #[doc(hidden)]
-        enum __Field #lifetime {
+        enum #this_type #lifetime {
```

Comment:
> I think this is still semantically not correct because `this_type` is normally allowed to include generic parameters (the ones that the user wrote in their `remote` attribute).

Hm, did you mean, that almost all places where `#this_type` is used, is incorrect? Because in all cases except one it is written as `#this_type #ty_generics`. The only place where lone `#this_type` is used is
https://github.com/serde-rs/serde/blob/25381be0c93f32aa43ce0e399c45442965e81ecc/serde_derive/src/de.rs#L423-L426

> Overall I would prefer to keep `__Field` as originally written everywhere in this file.
Ok, I remove this commit

Reply:
`#this_type #ty_generics` has a dedicated check to ensure that at most one of `this_type` or `ty_generics` actually contains any generics. https://github.com/serde-rs/serde/blob/25381be0c93f32aa43ce0e399c45442965e81ecc/serde_derive/src/internals/check.rs#L20-L40

To illustrate the difference between `#this_type #ty_generics` vs the `enum #this_type #lifetime` that was written here, consider that all of the existing usages would work if you had this_type=`Foo<()>`, whereas this usage is different and that is not a meaningful thing to write here.

Reply:
> To illustrate the difference between `#this_type #ty_generics` vs the `enum #this_type #lifetime` that was written here, consider that all of the existing usages would work if you had this_type=`Foo<()>`, whereas this usage is different and that is not a meaningful thing to write here.

Quite a hard to do that because I do not see obvious link between the check that you pointed out and the content of `this_type` and `ty_generics`. I will take your word that everything is in fine here.